### ISSUE TITLE
responsive div with multiple paragraphs - text vanishes at certain widths

### ISSUE THREAD

**Issue by anon1970** [BeSite/jQuery.dotdotdot#75]
My HTML is this:
```
	<body>
		<div class="box" id="dot5">
			<p>Dot5 was originally an example of an [+]/[-] box,
			but has now been adapted to include resizing.
			When closed, the box recalculates and repositions the ellipsis.
			When open, all text is shown and the box size changes to suit.</p>
			<p>Lorem Ipsum is simply dummy text of the printing and typesetting industry.</p>
		</div>
	</body>
```

and CSS styling is this:
```
			div.box {
				border: 1px solid #ccc;
				height: 60px;
				padding: 15px;
				/* overflow: hidden; */
			}
```
and using the sample index.html with only minor changes:
```
		<script type="text/javascript" language="javascript">
			$(function() {
				var $dot5 = $('#dot5');
				$dot5.append(' <a class="toggle" href="#"><span class="open">More</span><span class="close">Less</span></a>');
				
				function createDots() {
					$dot5.dotdotdot({
						after: 'a.toggle',
						watch: 'window'
					});
				}
				function destroyDots() {
					$dot5.trigger('destroy');
				}
				createDots();

				$dot5.on('click', 'a.toggle',
					function() {
						$dot5.toggleClass('opened');
						if ($dot5.hasClass('opened')) {
							destroyDots();
						} else {
							createDots();
						}
						return false;
					}
				);
			});
		</script>
```
then I find that resizing the browser window causes the HTML text to vanish at certain narrow browser widths.

This seems to happen only when the "More" tag would have been placed somewhere near the junction of the two paragraphs. i.e. near
`
 to suit.</p><p>Lorem Ipsum
`
Note that the div height is specified in the CSS.

Is this a bug?

**Comment by FrDH** [comment]
Is there an online example?

**Comment by anon1970** [comment]
Try this:
http://codepen.io/anon/pen/doVdBo
All the text within the div vanishes when the browser window width is greater than 534px.

**Comment by jtag05** [comment]
I'm also experiencing this issue. Has any progress been made for a workaround, or a solution implemented in the library?

**Comment by jtag05** [comment]
I've done a little testing and find this is directly related to using the 'after' attribute in the config object.

**Comment by FrDH** [comment]
I don't know why, but for some reason the "after" element goes bananas when used in combination with HTML markup.
- Replace the `<p>` tags with pure text -> no issue
- Remove the "after" option -> no issue
- Use the "after" option and have `<p>` tags -> issue...

**Comment by jtag05** [comment]
I have more or less traced it to a single block of code that decides if you use the .after() or .append() methods to add the 'after' element (line: 376). 

The issue only occurs when the end of a <p> element coincides with the designated height for plugin.

I haven't had time to test further and simply did a .replace() on the content before initializing the plugin as a workaround. 

My thinking is leaning in the direction that the append() method is being called on the wrong element, somehow causing the bug to occur.

**Comment by AlexanderLea** [comment]
Hi - I'm also experiencing this issue. Is there any suggestion/progress on a fix for this? I'd like to keep the styling of paragraphs intact, so I'd rather not remove them. 

This also appears to be an issue if you replace the <p>s with <span>s and use any kind of new line styling "float:left;clear:left;" or "display:block".

**Comment by dhunnapotha** [comment]
I hope [this pull request](https://github.com/BeSite/jQuery.dotdotdot/pull/93) fixes the issue.

**Comment by narcolepticinsomniac** [comment]
I'm having a similar issue, but the whole child element is vanishing. I'm reluctant to open a new issue, because I've had no luck creating a reduced test case. If anyone's willing to test a userscript and userstyle combo, I'd really appreciate any advice, because I'm getting nowhere with it. Seems like a bug to me, but mot user error does till someone tells you where you went wrong.

The function used to work perfectly with dotdotdot set to ellipse the child elements, like:

```
$('.card-title > a, .card-full-right > a').dotdotdot({
    height: "watch",
    truncate: "letter",
    watch: "true"
});
```

For whatever reason, recently that code started removing all text, leaving just the ellipsis, so I switched it to act on the parent `$('.card-title, .card-full-right')`, which seemed to work. Problem is, responsiveness is all messed up. The child elements are randomly removed at different viewport dimensions. Whenever the text does pop back in, the ellipses are correctly applied.

[Screen record](https://streamable.com/3tnp2)

[userscript](https://pastebin.com/tbrp2Lpg)

[userstyle](https://pastebin.com/R8x5FvGf)

[test URL](https://bit.ly/2Lr8UaX)

Elements in question are in the final slide of the slideshow.

**Comment by narcolepticinsomniac** [comment]
My bad, posted in the wrong repo. =)

**Comment by narcolepticinsomniac** [comment]
I'm having a similar issue, but the whole child element is vanishing. I'm reluctant to open a new issue, because I've had no luck creating a reduced test case. If anyone's willing to test a userscript and userstyle combo, I'd really appreciate any advice, because I'm getting nowhere with it. Seems like a bug to me, but most user error does till someone tells you where you went wrong.

The function used to work perfectly with dotdotdot set to ellipse the child elements, like:

```
$('.card-title > a, .card-full-right > a').dotdotdot({
    height: "watch",
    truncate: "letter",
    watch: "true"
});
```

For whatever reason, recently that code started removing all text, leaving just the ellipsis, so I switched it to act on the parent `$('.card-title, .card-full-right')`, which seemed to work. Problem is, responsiveness is all messed up. The child elements are randomly removed at different viewport dimensions. Whenever the text does pop back in, the ellipses are correctly applied.

[Screen record](https://streamable.com/3tnp2)

[userscript](https://pastebin.com/tbrp2Lpg)

[userstyle](https://pastebin.com/R8x5FvGf)

[test URL](https://bit.ly/2Lr8UaX)

Elements in question are in the final slide of the slideshow.
